File: frontend/lib/user/model/user.config.dart
/// This class defines user configuration options per user
class UserConfig {
  /// If we should hide balances on the users display
  bool privateMode;

  /// The net worth range to display by default
  String netWorthRange;

  /// A URL set by the frontend if sprout is running in non web mode
  String? connectionUrl;

  UserConfig({this.privateMode = false, this.netWorthRange = "oneDay", this.connectionUrl});

  Map<String, dynamic> toJson() {
    return {'privateMode': privateMode, 'netWorthRange': netWorthRange};
  }

  factory UserConfig.fromJson(Map<String, dynamic> json) {
    return UserConfig(privateMode: json['privateMode'] ?? false, netWorthRange: json['netWorthRange'] ?? "oneDay");
  }
}

File: backend/src/user/model/user.config.ts
import { DatabaseDecorators } from "@backend/database/decorators";
import { DatabaseBase } from "@backend/database/model/database.base";

/**
 * This class defines user configuration options per user
 */
@DatabaseDecorators.entity()
export class UserConfig extends DatabaseBase {
  /** If we should hide balances on the users display */
  @DatabaseDecorators.column({ nullable: false, default: false })
  privateMode: boolean;

  /** The net worth range to display by default */
  @DatabaseDecorators.column({ nullable: false, default: "oneDay" })
  netWorthRange: string;

  constructor(privateMode: boolean, netWorthRange: string) {
    super();
    this.privateMode = privateMode;
    this.netWorthRange = netWorthRange;
  }
}
